### PR TITLE
Add server version aliasing feature

### PR DESCRIPTION
**Pull request by willbroadbelt**
Adding the server aliasing feature that was discussed. It will allow cbdyncluster users to specify a version like '6.6-stable', '6.5-release', etc. which will return a build version i.e 6.6.0-7813, or a release version i.e. 6.5.1, that will be used as the cluster version. (Note that the x.x-**stable** tag can return either a build version, or a normal release version, but cbdynclusterd interprets this correctly either way).

I currently just have the products/aliasing file in my own repo (https://github.com/willbroadbelt/cb-alias), though this should probably be moved to a couchbaselabs repo at some point.

### CODE REVIEW

File: daemon/alias.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+//Clones/pulls the github alias repo
+func GetConfigRepo() error {
+	_, err := git.PlainClone(helper.AliasRepoPath, false, &git.CloneOptions{
+		URL:               helper.AliasRepo,
+		Progress:          os.Stdout,
+	})
+
+	if err == git.ErrRepositoryAlreadyExists {
```
Comment (chvck) [inline daemon/alias.go:44]: Not necessary but errors.Is(err, git.ErrRepositoryAlreadyExists) is a better way of doing this as of 1.13

File: daemon/nodes.go [hunk 2]
```diff
@@ -107,6 +107,39 @@ func parseServerVersion(version string) (*NodeVersion, error) {
...
+	}
+
+	p, err := GetProductsMap()
+	if err != nil {
+		return "", err
+	}
+
+	var serverBuild string
+	if buildParts[1] == "release" {
+		serverBuild = p["couchbase-server"][buildParts[0]].Release
```
Comment (chvck) [inline daemon/nodes.go:129]: This is gonna go bigly bang if someone requests a version that isn't in the alias map.

File: daemon/alias.go (hunk 1 above)
Comment (willbroadbelt) [reply daemon/alias.go:44]: Ah cool, I'll update this.

File: daemon/nodes.go (hunk 2 above)
Comment (willbroadbelt) [reply daemon/nodes.go:129]: It works fine. If there is no map value, serverBuild is still an empty string which is then handled below and you get back e.g. "No build version found for 8.8-release".

**Pull request merged**